### DISCUSSION TITLE
Progress?

### DISCUSSION THREAD

**Discussion by DynasB4Vaginas** [tarxvftech/FCS-152#74] [General]
Has there been any progress made on this radio?? Specifically it's ability to interface w TAK?

**Comment by tarxvftech** [comment]
No, and it will not generally be useful for any TAK-like so don't hold your breath.

The radio board in it, and the main board connections to that board, are so limited that you only get what it says on the tin. There is nearly no benefit to hackability other than UX and management improvements. Further development would require custom hardware.

**Reply by DynasB4Vaginas** [reply]
Ahhh that's a bummer.. I'm not super fermilar with any of this stuff but just have a basic understanding of some of it. With that being said, pls excuse my ignorance, but what do you mean by UX and management improvements? Lol Is that referring to improvements via coding or?? As far as new hardware goes, I'm assuming you mean like a drop in replacement for the internal chips/motherboard?? 

Also being that it utilizes an ESP32 chip which (correct me if I'm wrong) is the same chip used in the LoRa meshtastic boards, would there be any way to possibly convert it to a meshtastic device? I'm obviously not even remotely smart enough to understand how or what would be needed to make the conversion but just wondering if that's something that could potentially be done? 

I guess what I'm asking is if there is any potential future possibility of expanding the use or capabilities of these radios? Obviously I'm not expecting anything even remotely close to the genuine prc but I ask because I have been wanting one for a while(just a living room/neighborhood  larper) and have been having a ridiculously hard time trying to decide which one of the repros to get out of the FCS/TCA/TRI. I've been really really leaning towards FCS even tho it's currently the least capable of the 3 options specifically since its open source and in hopes that somebody will develope something to expand its capabilities.. frankly, with such a hugeeeee market for such a thing out there I'm honestly surprised nobody has come up with anything yet... 

Obviously you know WAYYYYY more about any of this than I do so what would you do in my shoes? Idkk if youre fermilar with the TCA/TRI radios(if youre not then no worries) but if you are and you had to pick 1 of the 3, knowing what you know about them, which would you recommend??

**Reply by tarxvftech** [reply]
Ignorance is alright, and asking questions is the only way to get out of that!


### ESP32
Meshtastic is effectively an application on top of LoRa, a digital RF protocol.
An ESP32 is the main chip on these FCS radios and most (all?) of the meshtastic boards.


That, however, does not mean the ESP32 is the chip that does LoRa.
ESP32 is a wide range of hardware, but the only radios they come with are (almost always) WiFi, and (most of the time) Bluetooth.

The Meshtastic devices have a LoRa specific chip made by, if memory serves, Semtech. That LoRa chip is what actually does the radio parts - it's effectively a baseband processor and modem, amplifiers, filters, etc all in one chip. Very convenient compared to having to build that all onto a board yourself.

The FCS radios have an ESP32 doing the user interface, or HMI (human machine interface). That means it's the ESP32 that's drawing to the screen, taking input via buttons and the keypad, and that sort of thing. 
There's another chip on a second board, the RF (radio frequency) board, that's responsible for all the radio stuff, and due to the way that radio board is designed we'd basically have to design entirely new electronics - both boards! - and just reuse the case to make this radio capable of anything further than it already does.

### Possible improvements to FCS-152
When I said UX or management, I was thinking of things like fixing the way the knob on top is read (the current code from FCS misses 'clicks' or even goes the wrong way because of how it's written), or making it so channels could be programmed from a web browser. The core functionality of the radio itself would remain the same, only some minor 'nice-to-haves' could be added with only a software update.
Unfortunately that means it's more fun to work on other radios that don't have these limitations, so even these nice-to-haves probably won't come to the FCS-152 any time soon.

### Which PRC-152 look-alike to get:
Whichever has the features you want the most. Headset and accessory compatibility might be critical for you, or perhaps you want more supported bands. FCS is great choice if you are a programmer and want to experiment on a radio with a great aluminum case, but otherwise I believe it is lagging behind in terms of features to TCA and TRI. I would not recommend the FCS on the basis of future expandability, because no one is working on it.

You might be surprised how small the market actually is for these things, but at the same time it's clearly keeping three different PRC-152 clones (and really more, if you include the Baofeng plastic one) up and running. 

I'm actually not that familiar with the TCA or TRI options. I believe one had 220MHz, and as an amateur radio operator that would be useful to me, but the programmability is my true preference and I wanted to vote with my wallet and support FCS for open sourcing their code.

### Bottom line
In your shoes, I'd get a meshtastic device for TAK purposes and literally any voice radio that supports your desired accessories for voice purposes.
